### PR TITLE
Add server url parameter for internal github instances

### PR DESCRIPTION
**Pull request by jaleksynas**
Issue #1288

### CODE REVIEW

**Comment by jaleksynas** [comment]
@KrauseFx Not sure how to respond to hound line length violations.  I will address single vs. double quotes.  The other violations are pre-exisiting (redundant self?) or of a matching style to existing code (long lines).

File: lib/fastlane/actions/get_github_release.rb [hunk 1]
```diff
@@ -109,6 +110,15 @@ def self.available_options
                                          raise "Please only pass the path, e.g. 'KrauseFx/fastlane'".red if value.include? "github.com"
                                          raise "Please only pass the path, e.g. 'KrauseFx/fastlane'".red if value.split('/').count != 2
                                        end),
+          FastlaneCore::ConfigItem.new(key: :server_url,
+                                       env_name: "FL_GITHUB_RELEASE_SERVER_URL",
+                                       description: "The server url. e.g. 'https://your.github.server/api/v3' (Default: 'https://api.github.com')",
+                                       default_value: "https://api.github.com",
+                                       optional: true,
+                                       verify_block: proc do |value|
+                                         raise "Please don't include the trailing slash in the server url".red if value.end_with? "/"
```
Comment (i2amsam) [inline lib/fastlane/actions/get_github_release.rb]: How about about just stripping the extra /?

Comment (jaleksynas) [inline lib/fastlane/actions/get_github_release.rb]: Was just more code in more places, but i could do that if you'd prefer

File: lib/fastlane/actions/set_github_release.rb [hunk 2]
```diff
@@ -26,9 +26,10 @@ def self.run(params)
 
         repo_name = params[:repository_name]
         api_token = params[:api_token]
+        server_url = params[:server_url]
```
Comment (i2amsam) [inline lib/fastlane/actions/set_github_release.rb]:
```ruby
server_url = params[:server_url]

server_url = server_url[0..-2] if server_url.ends_with? '/'
```

File: lib/fastlane/actions/get_github_release.rb (hunk 1 above)
Comment (i2amsam) [inline lib/fastlane/actions/get_github_release.rb]: Thank you for this PR, this is a nice addition!  I made a suggestion for normalizing the form it from the params.

Comment (jaleksynas) [inline lib/fastlane/actions/get_github_release.rb]: Will do the normalizing code in both actions. Though may be cleaner and
more generally useful if ConfigItem could take a sanitize block similar to
the verify block in which values can be normalized up front.

**Comment by KrauseFx** [comment]
Sorry about Hound, seems like he's confused, you can ignore him as long as the validation works locally using `rubocop -a`

File: lib/fastlane/actions/set_github_release.rb (hunk 2 above)
Comment (KrauseFx) [inline lib/fastlane/actions/set_github_release.rb]: Why did you add this comment?

**Comment by KrauseFx** [comment]
This looks great, thanks for your contribution @jaleksynas :+1:

File: lib/fastlane/actions/set_github_release.rb (hunk 2 above)
Comment (i2amsam) [inline lib/fastlane/actions/set_github_release.rb]: @KrauseFx I was suggesting accepting `server_urls` with or without trailing `/`'s
